extensions/funqy/funqy-knative-events/runtime/src/main/java/io/quarkus/funqy/runtime/bindings/knative/events/VertxRequestHandler.java

```diff
@@ -214,7 +215,17 @@ private void execute(CloudEvent event, RoutingContext routingContext, ResponsePr
                                     ObjectWriter writer = (ObjectWriter) invoker.getBindingContext()
                                             .get(ObjectWriter.class.getName());
                                     httpResponse.putHeader("Content-Type", "application/json");
-                                    httpResponse.end(writer.writeValueAsString(obj));
+
+                                    if (obj instanceof CloudEventOutput) {
+                                        CloudEventOutput<?> cloudEventOutput = (CloudEventOutput<?>) obj;
+
+                                        httpResponse.putHeader("ce-source", cloudEventOutput.source());
+                                        httpResponse.putHeader("ce-type", cloudEventOutput.type());
+
+                                        httpResponse.end(writer.writeValueAsString(cloudEventOutput.data()));
```

Comment:
Should open a separate ticket to handle the case where generic information is not provided in the function signature.  I.e., if the function cloud event data is variable between calls.

Reply:
here is the issue https://github.com/quarkusio/quarkus/issues/14191